src/Region/RegionHandler.cc

```diff
@@ -318,24 +340,88 @@ bool RegionHandler::SetHistogramRequirements(
         return true;
     }
 
-    if (_regions.count(region_id)) {
-        // Save frame pointer
-        _frames[file_id] = frame;
+    if (!RegionSet(region_id)) {
+        spdlog::error("Histogram requirements failed: no region with id {}", region_id);
+        return false;
+    }
 
-        // Make HistogramConfig vector of requirements
-        std::vector<HistogramConfig> input_configs;
-        for (auto& config : configs) {
-            HistogramConfig hist_config(config.coordinate(), config.channel(), config.num_bins());
-            input_configs.push_back(hist_config);
-        }
+    if (!IsClosedRegion(region_id)) {
+        spdlog::debug("Histogram requirements not valid for region {} type", region_id);
+        return false;
+    }
 
-        // Set requirements
-        ConfigId config_id(file_id, region_id);
-        _histogram_req[config_id].configs = input_configs;
+    // Save frame pointer
+    _frames[file_id] = frame;
+
+    // Make HistogramConfig vector of requirements
+    std::vector<HistogramConfig> input_configs;
+    for (auto& config : configs) {
+        HistogramConfig hist_config(config.coordinate(), config.channel(), config.num_bins());
+        input_configs.push_back(hist_config);
```

Comment:
I don't know how big this loop is, but I would suggest using `emplace_back(...)`.

Reply:
For a region histogram, the only current setting is the stokes coordinate, so the loop is nstokes.